app/scripts/modules/core/src/pipeline/config/triggers/artifacts/expectedArtifact.component.ts

```diff
@@ -0,0 +1,50 @@
+import { IComponentController, IComponentOptions, module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IExpectedArtifact } from 'core/domain/IExpectedArtifact';
+import { IPipeline } from 'core/domain/IPipeline';
+import { IArtifact } from 'core/domain/IArtifact';
+
+class ExpectedArtifactController implements IComponentController {
+  public expectedArtifact: IExpectedArtifact;
+  public pipeline: IPipeline;
+
+  public removeArtifact(): void {
+    const artifactIndex = this.pipeline.expectedArtifacts.indexOf(this.expectedArtifact);
+    this.pipeline.expectedArtifacts.splice(artifactIndex, 1);
+
+    this.pipeline.triggers
+      .forEach(t => t.expectedArtifacts = t.expectedArtifacts.filter(a => !this.expectedArtifactEquals(a, this.expectedArtifact)));
+  }
+
+  private artifactEquals(first: IArtifact, other: IArtifact): boolean {
+    return (!first && !other)
```

Comment:
This is a bummer. Is this every propert of the artifact? I think jake did this because the objects had some angular bits mixed in - I forgot to mention that there’s an `angular.equals` that ignores those bits.

Reply:
`angular.equals` sounds great